server/repository/src/db_diesel/encounter_row.rs

```diff
@@ -33,6 +33,9 @@ table! {
     }
 }
 
+joinable!(encounter -> program (program_id));
+allow_tables_to_appear_in_same_query!(encounter, program);
```

Comment:
you mean putting them into the big list? or do you mean just collect them there?

I am a bit concerned that `allow_tables_to_appear_in_same_query` generates too much code because it creates an entry for every combination of tables. The generated code for each combination is quite small though. If I did the math right: with 25 tables it generates 325 entries (for 30 tables it would already generate 465 entries).

Otherwise, I like having them straight next to the table definition because then you can see the columns it references, but not too strong about it...

Reply:
Oki, i don't really mind, I had a couple of use case where one list was easier, i think it was with `name` table, but can't actually remember. I didn't think of the number of generated entries, looks like there are 2031 generated lines (3 lines for each impl) in the combined `allow_tables_to_appear_in_same_query` (I guess makes sense to minimise that if possible)